pkg/resources/grant_application_role.go

```diff
@@ -127,7 +121,7 @@ func ReadContextGrantApplicationRole(ctx context.Context, d *schema.ResourceData
 	switch objectType {
 	case sdk.ObjectTypeRole:
 		{
-			if _, err := client.Roles.ShowByID(ctx, sdk.NewShowByIdRoleRequest(sdk.NewAccountObjectIdentifierFromFullyQualifiedName(targetIdentifier))); err != nil && errors.Is(err, sdk.ErrObjectNotExistOrAuthorized) {
+			if _, err := client.Roles.ShowByID(ctx, sdk.NewAccountObjectIdentifierFromFullyQualifiedName(targetIdentifier)); err != nil && errors.Is(err, sdk.ErrObjectNotExistOrAuthorized) {
```

Comment:
the error check should be `errors.Is(err, sdk.ErrObjectNotFound)` (also in line 136; cannot comment there ;_;)

Reply:
Fixed in both lines.